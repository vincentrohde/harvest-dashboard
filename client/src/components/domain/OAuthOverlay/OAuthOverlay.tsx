// Libs
import React from 'react';
import {Button} from 'semantic-ui-react';

// Styles
import './OAuthOverlay.scss';

const OAuthOverlay = () => {
    const openLink = (event: React.MouseEvent) => {
        const target = event.target as HTMLElement;
        const ctaButton = target.closest('.js-cta-link');

        if (ctaButton && ctaButton instanceof HTMLElement) {
            const link = ctaButton.dataset.link;
            if (link) {
                window.location.href = link;
            }
        }
    };
    return (
        <div className="OAuthOverlay">
            <h2 className="title">Authorization required üîê</h2>
            <div className="section">
                <p className="notice">
                    To view and edit your Harvest time entries on this website, you will have to
                    allow us to access your account. This is done through the OAuth workflow
                    provided by Harvest.
                    <br />
                    <br />
                    The link below takes you to the official Harvest website, where you can
                    authorize us to request access tokens for your account. They will then be sent
                    back to you, encoded. During your session, the website will decode them to
                    access your account. The tokens are only stored inside your browser, as cookies.
                    <br />
                    <br />
                    For additional info on the process, feel free to visit our GitHub repository üêô
                    <a
                        className="repo-link"
                        href="https://github.com/vincentrohde/harvest-dashboard"
                    >
                        https://github.com/vincentrohde/harvest-dashboard
                    </a>
                </p>
                <Button
                    className="cta-link js-cta-link"
                    content={
                        <a
                            href={`https://id.getharvest.com/oauth2/authorize?client_id=${process.env.OAUTH_APP_ID}&response_type=code`}
                        >
                            Authorize application
                        </a>
                    }
                    icon="right arrow"
                    onClick={openLink}
                    data-link={`https://id.getharvest.com/oauth2/authorize?client_id=${process.env.OAUTH_APP_ID}&response_type=code`}
                    primary
                    labelPosition="right"
                />
            </div>

            <div className="section">
                <h3 className="section-title">Create new account ‚è∞</h3>
                <p className="notice">
                    If you currently have no Harvest account, you can easily create one for
                    yourself. The link below will take you to the official Harvest website, where
                    you can register a new personal account.
                </p>
                <Button
                    className="cta-link js-cta-link"
                    content={<a href="https://www.getharvest.com">Create new account</a>}
                    onClick={openLink}
                    data-link="https://www.getharvest.com"
                    icon="right arrow"
                    primary
                    labelPosition="right"
                />
            </div>
        </div>
    );
};

export default OAuthOverlay;
