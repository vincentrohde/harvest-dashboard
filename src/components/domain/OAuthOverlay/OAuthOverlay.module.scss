.OAuthOverlay {
    .title, .section-title {
        margin-bottom: 1.5rem;
    }

    .section {
        margin-bottom: 6rem;

        .section-title {
            font-size: 1.4rem;
        }

        .notice {
            font-size: 1.2rem;
            line-height: 1.5;
        }

        .repo-link {
            display: block;
        }

        .cta-link {
            margin-top: 1rem;

            a {
                color: #fff;
            }
        }
    }
}
