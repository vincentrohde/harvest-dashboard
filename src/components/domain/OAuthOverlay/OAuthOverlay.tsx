// Libs
import React from 'react';
import { Button } from 'semantic-ui-react';
import classNames from 'classnames';

// Styles
import styles from './OAuthOverlay.module.scss';

const OAuthOverlay = () => {
    const openLink = (event: React.MouseEvent) => {
        const target = event.target as HTMLElement;
        const ctaButton = target.closest('.js-cta-link');

        if (ctaButton && ctaButton instanceof HTMLElement) {
            const link = ctaButton.dataset.link;
            if (link) {
                window.location.href = link;
            }
        }
    };
    return (
        <div className={styles.OAuthOverlay} data-test="oauth-overlay">
            <h2 className={styles.title}>Authorization required üîê</h2>
            <div className={styles.section}>
                <p className={styles.notice}>
                    To view and edit your Harvest time entries on this website, you will have to
                    allow us to access your account. This is done through the OAuth workflow
                    provided by Harvest.
                    <br />
                    <br />
                    The link below takes you to the official Harvest website, where you can
                    authorize us to request access tokens for your account. They will then be sent
                    back to you, encoded. During your session, the website will decode them to
                    access your account. The tokens are only stored inside your browser, as cookies.
                    <br />
                    <br />
                    For additional info on the process, feel free to visit our GitHub repository üêô
                    <a
                        className={styles['repo-link']}
                        href="https://github.com/vincentrohde/harvest-dashboard"
                    >
                        https://github.com/vincentrohde/harvest-dashboard
                    </a>
                </p>
                <Button
                    className={classNames(styles['cta-link'], 'js-cta-link')}
                    content={
                        <a
                            href={`https://id.getharvest.com/oauth2/authorize?client_id=${process.env.OAUTH_APP_ID}&response_type=token`}
                        >
                            Authorize application
                        </a>
                    }
                    icon="right arrow"
                    onClick={openLink}
                    data-link={`https://id.getharvest.com/oauth2/authorize?client_id=${process.env.OAUTH_APP_ID}&response_type=token`}
                    data-test="authorize-app"
                    primary
                    labelPosition="right"
                />
            </div>

            <div className={styles.section}>
                <h3 className={styles['section-title']}>Create new account ‚è∞</h3>
                <p className={styles.notice}>
                    If you currently have no Harvest account, you can easily create one for
                    yourself. The link below will take you to the official Harvest website, where
                    you can register a new personal account.
                </p>
                <Button
                    className={classNames(styles['cta-link'], 'js-cta-link')}
                    content={<a href="https://www.getharvest.com">Create new account</a>}
                    onClick={openLink}
                    data-link="https://www.getharvest.com"
                    data-test="create-account"
                    icon="right arrow"
                    primary
                    labelPosition="right"
                />
            </div>
        </div>
    );
};

export default OAuthOverlay;
