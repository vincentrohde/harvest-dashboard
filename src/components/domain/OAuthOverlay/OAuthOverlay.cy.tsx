// Libraries
import React from 'react';

// Components
import OAuthOverlay from './OAuthOverlay';

describe('<OAuthOverlay />', () => {
    it('authorize button exists and links correctly', () => {
        cy.mount(<OAuthOverlay />)
        cy.get('[data-test="authorize-app"]')
            .contains("a")
            .should("have.attr", "href", `https://id.getharvest.com/oauth2/authorize?client_id=${Cypress.env('OAUTH_APP_ID')}&response_type=token`);
    });

    it('create new account button exists and links correctly', () => {
        cy.mount(<OAuthOverlay />)
        cy.get('[data-test="create-account"]')
            .contains("a")
            .should("have.attr", "href", "https://www.getharvest.com");
    });
})
