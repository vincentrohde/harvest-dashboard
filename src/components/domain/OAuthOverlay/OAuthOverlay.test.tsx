import { render } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

import OAuthOverlay from './OAuthOverlay';

describe('<OAuthOverlay />', () => {
    it('renders OAuthOverlay correctly', () => {
        const { asFragment } = render(<OAuthOverlay />);

        expect(asFragment()).toMatchSnapshot();
    });
})
