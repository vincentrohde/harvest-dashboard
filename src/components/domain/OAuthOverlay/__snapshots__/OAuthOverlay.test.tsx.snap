// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cancel /> renders OAuthOverlay correctly 1`] = `
<DocumentFragment>
  <div
    class="OAuthOverlay"
    data-test="oauth-overlay"
  >
    <h2
      class="title"
    >
      Authorization required üîê
    </h2>
    <div
      class="section"
    >
      <p
        class="notice"
      >
        To view and edit your Harvest time entries on this website, you will have to allow us to access your account. This is done through the OAuth workflow provided by Harvest.
        <br />
        <br />
        The link below takes you to the official Harvest website, where you can authorize us to request access tokens for your account. They will then be sent back to you, encoded. During your session, the website will decode them to access your account. The tokens are only stored inside your browser, as cookies.
        <br />
        <br />
        For additional info on the process, feel free to visit our GitHub repository üêô
        <a
          class="repo-link"
          href="https://github.com/vincentrohde/harvest-dashboard"
        >
          https://github.com/vincentrohde/harvest-dashboard
        </a>
      </p>
      <button
        class="ui icon primary right labeled button cta-link js-cta-link"
        data-link="https://id.getharvest.com/oauth2/authorize?client_id=undefined&response_type=token"
      >
        <i
          aria-hidden="true"
          class="right arrow icon"
        />
        <a
          href="https://id.getharvest.com/oauth2/authorize?client_id=undefined&response_type=token"
        >
          Authorize application
        </a>
      </button>
    </div>
    <div
      class="section"
    >
      <h3
        class="section-title"
      >
        Create new account ‚è∞
      </h3>
      <p
        class="notice"
      >
        If you currently have no Harvest account, you can easily create one for yourself. The link below will take you to the official Harvest website, where you can register a new personal account.
      </p>
      <button
        class="ui icon primary right labeled button cta-link js-cta-link"
        data-link="https://www.getharvest.com"
      >
        <i
          aria-hidden="true"
          class="right arrow icon"
        />
        <a
          href="https://www.getharvest.com"
        >
          Create new account
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<OAuthOverlay /> renders OAuthOverlay correctly 1`] = `
<DocumentFragment>
  <div
    class="OAuthOverlay"
    data-test="oauth-overlay"
  >
    <h2
      class="title"
    >
      Authorization required üîê
    </h2>
    <div
      class="section"
    >
      <p
        class="notice"
      >
        To view and edit your Harvest time entries on this website, you will have to allow us to access your account. This is done through the OAuth workflow provided by Harvest.
        <br />
        <br />
        The link below takes you to the official Harvest website, where you can authorize us to request access tokens for your account. They will then be sent back to you, encoded. During your session, the website will decode them to access your account. The tokens are only stored inside your browser, as cookies.
        <br />
        <br />
        For additional info on the process, feel free to visit our GitHub repository üêô
        <a
          class="repo-link"
          href="https://github.com/vincentrohde/harvest-dashboard"
        >
          https://github.com/vincentrohde/harvest-dashboard
        </a>
      </p>
      <button
        class="ui icon primary right labeled button cta-link js-cta-link"
        data-link="https://id.getharvest.com/oauth2/authorize?client_id=undefined&response_type=token"
      >
        <i
          aria-hidden="true"
          class="right arrow icon"
        />
        <a
          href="https://id.getharvest.com/oauth2/authorize?client_id=undefined&response_type=token"
        >
          Authorize application
        </a>
      </button>
    </div>
    <div
      class="section"
    >
      <h3
        class="section-title"
      >
        Create new account ‚è∞
      </h3>
      <p
        class="notice"
      >
        If you currently have no Harvest account, you can easily create one for yourself. The link below will take you to the official Harvest website, where you can register a new personal account.
      </p>
      <button
        class="ui icon primary right labeled button cta-link js-cta-link"
        data-link="https://www.getharvest.com"
      >
        <i
          aria-hidden="true"
          class="right arrow icon"
        />
        <a
          href="https://www.getharvest.com"
        >
          Create new account
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
